### PR TITLE
Fix issue 1414 remove hard limit on ft.search

### LINKED ISSUE #1414: Configurable maximum results (PM-737)
The maximum number of results in RediSearch for FT.SEARCH is hard coded to 1 million.  In certain scenario's, this might be too limited and hence a module configuration is required  where `-1` would denote all results.

### PR DESCRIPTION
**Pull request by ashtul**
fix #1414

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -121,6 +121,23 @@ CONFIG_GETTER(getMaxDocTableSize) {
...
+  if (newsize == -1) {
+    newsize = UINT64_MAX;
+  }
+  config->maxSearchResults = newsize;
+  return REDISMODULE_OK;
+}
+
+CONFIG_GETTER(getMaxSearchResults) {
+  sds ss = sdsempty();
+  return sdscatprintf(ss, "%lu", config->maxSearchResults);
```
Comment (MeirShpilraien) [inline src/config.c:138]: So if we will set it to -1 and we will do ft.configget we will not see -1 no?

File: src/pytest/test_config.py [hunk 2]
```diff
@@ -156,3 +157,7 @@ def test_arg_str(arg_name, arg_value, ret_value=None):
     test_arg_str('ON_TIMEOUT', 'fail')
     test_arg_str('PARTIAL_INDEXED_DOCS', '0', 'false')
     test_arg_str('PARTIAL_INDEXED_DOCS', '1', 'true')
+    test_arg_str('MAXSEARCHRESULTS', '100', '100')
+    test_arg_str('MAXSEARCHRESULTS', '-1', '18446744073709551615')
```
Comment (ashtul) [inline src/pytest/test_config.py:161]: @MeirShpilraien 
This is what you will see...

File: src/config.c (hunk 1 above)
Comment (ashtul) [reply src/config.c:141]: was uint_max. changed to unlimited

File: src/pytest/test_config.py (hunk 2 above)
Comment (ashtul) [reply src/pytest/test_config.py]: Changed to unlimited

**Pull request merged**